tests/test_tone_classification.py

```diff
@@ -0,0 +1,163 @@
+    if f0_non_nan.size != 0:
+        tone = np.array([])
+        # below calculation is from Table 1 of Learning tone distinctions for Mandarin Chinese paper
+        exp_T1_Fb = fh
+        exp_T2_Fb = fh*np.sqrt(0.5)-fh/2*1/12
+        exp_T3_Fb = fh*np.sqrt(0.5)
+        exp_T4_Fb = fh+fh/2*2/12
+        act_f0_b = f0_non_nan[1]
+        exp_tone_Fb = [exp_T1_Fb, exp_T2_Fb, exp_T3_Fb, exp_T4_Fb]
+        tone = np.concatenate((tone, find_tones(exp_tone_Fb, act_f0_b)))
+        
+        # print('Expected T1 beginning freq: ', exp_T1_Fb)
+        # print('Expected T2 beginning freq: ', exp_T2_Fb)
+        # print('Expected T3 beginning freq: ', exp_T3_Fb)
+        # print('Expected T4 beginning freq: ', exp_T4_Fb)
+        # print('Actual f0 beginning value: ', act_f0_b, '\n')
+        
+        # below calculation is from Table 1 of Learning tone distinctions for Mandarin Chinese paper
+        exp_T1_Fe = exp_T1_Fb
+        exp_T2_Fe = fh+fh/2*2/12
+        exp_T3_Fe = exp_T3_Fb
+        exp_T4_Fe = 0.5*(fh+fh/2*2/12)
+        act_f0_e = f0_non_nan[-1]
+        exp_tone_Fe = [exp_T1_Fe, exp_T2_Fe, exp_T3_Fe, exp_T4_Fe]
+        tone = np.concatenate((tone, find_tones(exp_tone_Fe, act_f0_e)))
+
+        # print('Expected T1 end freq: ', exp_T1_Fe)
+        # print('Expected T2 end freq: ', exp_T2_Fe)
+        # print('Expected T3 end freq: ', exp_T3_Fe)
+        # print('Expected T4 end freq: ', exp_T4_Fe)
+        # print('Actual f0 end value: ', act_f0_e, '\n')
+        
+        # below calculation is from Table 1 of Learning tone distinctions for Mandarin Chinese paper
+        exp_T1_Fi = (exp_T1_Fb + exp_T1_Fe)/2
+        exp_T2_Fi = (exp_T2_Fb + exp_T2_Fe)/2
+        exp_T3_Fi = fh*0.5 - fh/2*3/12
+        exp_T4_Fi = (exp_T4_Fb + exp_T4_Fe)/2
+        act_f0_i = f0_non_nan[len(f0_non_nan)//2]
+        exp_tone_Fi = [exp_T1_Fi, exp_T2_Fi, exp_T3_Fi, exp_T4_Fi]
+        tone = np.concatenate((tone,find_tones(exp_tone_Fi, act_f0_i)))
+
+        # print('Expected T1 middle freq: ', exp_T1_Fi)
+        # print('Expected T2 middle freq: ', exp_T2_Fi)
+        # print('Expected T3 middle freq: ', exp_T3_Fi)
+        # print('Expected T4 middle freq: ', exp_T4_Fi)
+        # print('Actual f0 middle value: ', act_f0_i, '\n')
+        
+        tone = tone.astype(int)
+        count = np.bincount(tone)
+        # if the count is equal to 3, all estimated Fb, Fi and Fe values are close to the actual values
+        if count[count.argmax()] == 3:
+            final_tone = np.array([count.argmax()+1])
+        # if the count is equal to 2, some of the estimated Fb, Fi and Fe values are close to the actual values
+        else:
+            final_tone = np.where(count == 2)
+            final_tone = np.array([i+1 for i in final_tone])[0]
+        print(count)
+        print('Estimated tone: Tone ' + str(final_tone) + '\n')
+    else: 
+        final_tone = np.array([0])
+        print('all f0 estimation is nan \n')
+    return final_tone
```

Comment:
OOOH wait i understand now, this is combining both the function code and the test code into a single file. ok, i'll merge everything else in and then we can move the actual function to pitch_process, & put the unit tests for them in this file